File: backend/src/main/java/com/stefanini/taskmanager/domain/Task.java
package com.stefanini.taskmanager.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 150)
    private String titulo;

    @Column(length = 1000)
    private String descricao;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private TaskStatus status;

    @Column(name = "data_criacao", nullable = false, updatable = false)
    private OffsetDateTime dataCriacao;

    @PrePersist
    void onCreate() {
        if (dataCriacao == null) {
            dataCriacao = OffsetDateTime.now();
        }
        if (status == null) {
            status = TaskStatus.PENDENTE;
        }
    }
}

File: frontend/src/types/task.ts
export type TaskStatus = 'PENDENTE' | 'EM_ANDAMENTO' | 'CONCLUIDO';

export interface Task {
  id: number;
  titulo: string;
  descricao: string | null;
  status: TaskStatus;
  dataCriacao: string;
}

export interface CreateTaskRequest {
  titulo: string;
  descricao?: string;
  status?: TaskStatus;
}

export const STATUS_LABELS: Record<TaskStatus, string> = {
  PENDENTE: 'Pendente',
  EM_ANDAMENTO: 'Em andamento',
  CONCLUIDO: 'Concluído',
};
